src/main/java/com/shopping_app/shoppingApp/config/JWT/JwtTokenProvider.java

```diff
@@ -0,0 +1,54 @@
+package com.shopping_app.shoppingApp.config.JWT;
+
+import java.util.Base64;
+import java.util.Date;
+import javax.annotation.PostConstruct;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__({@Autowired}))
+@Slf4j
+public class JwtTokenProvider {
+
+    private final UserDetailsService userDetailsService;
+
+    @Value("${jwt.secret-key}")
+    private String secretKey;
+    private static final long EXPIRE_TIME = (1000 * 60 * (60*24));
+
+    @PostConstruct
+    protected void init() {
+        secretKey = Base64.getEncoder().encodeToString(secretKey.getBytes());
+    }
+
+    public String createToken(String email) {
+        Date now = new Date();
+        return Jwts.builder()
+                .setSubject(email)
+                .setIssuedAt(now)
+                .setExpiration(new Date(now.getTime() + EXPIRE_TIME))
+                .signWith(SignatureAlgorithm.HS256, secretKey).compact();
+    }
+
+    public Authentication getAuthentication(String token) {
+        String email = Jwts.parser().setSigningKey(secretKey).parseClaimsJws(token).getBody().getSubject();
+        UserDetails userDetails = userDetailsService.loadUserByUsername(email);
+        return new UsernamePasswordAuthenticationToken(userDetails, "", userDetails.getAuthorities());
+    }
+
+    public boolean validateToken(String token) {
+        Jwts.parser().setSigningKey(secretKey).parseClaimsJws(token).getBody();
+        return true;
+    }
```

Comment:
Can you explain how this `validateToken` function works?

Reply:
yeah  , if we pass valid token  this method which will not through any exeception and return true but if we pass invalid token (expired token ,or wrong token , etc.. ) then exception will throw and i handled this exception in "JwtTokenFilter" class and will not allow to proceed further